src/EditorFeatures/Core/Implementation/Formatting/FormatCommandHandler.FormatDocument.cs

```diff
@@ -93,13 +92,16 @@ public bool ExecuteCommand(FormatDocumentCommandArgs args, CommandExecutionConte
                     {
                         if (docOptions.GetOption(CodeCleanupOptions.AreCodeCleanupRulesConfigured))
                         {
-                            // Code cleanup
-                            var oldDoc = document;
-                            var codeCleanupChanges = GetCodeCleanupAndFormatChangesAsync(document, codeCleanupService, cancellationToken).WaitAndGetResult(cancellationToken);
-
-                            if (codeCleanupChanges != null && codeCleanupChanges.Count() > 0)
+                            using (context.OperationContext.AddScope(allowCancellation: true, EditorFeaturesResources.This_can_take_several_seconds_when_there_are_many_changes))
```

Comment:
i'm a little confused why we're not actually just properly doing progress here.  We have a context object we can report progress to, and we know the set of items we're going to execute as part of code-cleanup.  So we can both accurately give a progress bar, and also tell the user what we're actually running.  This would be better and would convey more info than just a generic "this may take a while".

Also, it will help people report issues to us.  For example, if we had this, then we would have seen immediately "oh... removing 'this.' takes a ***long*** time'".

Reply:
Note: i have a PR out for this here: https://github.com/dotnet/roslyn/pull/28353

However, it's not working for reasons that elude me.  I've pulled in @olegtk  to help out here.